pkg/operator/azurerm/operator.go

```diff
@@ -0,0 +1,117 @@
+package azurerm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/resources/armresources"
+
+	"github.com/seal-io/walrus/pkg/dao/model"
+	"github.com/seal-io/walrus/pkg/dao/types"
+	"github.com/seal-io/walrus/pkg/dao/types/status"
+	"github.com/seal-io/walrus/pkg/operator/azurerm/resourcestatus"
+	aztypes "github.com/seal-io/walrus/pkg/operator/azurerm/types"
+	optypes "github.com/seal-io/walrus/pkg/operator/types"
+	"github.com/seal-io/walrus/utils/hash"
+)
+
+const OperatorType = "AzureRM"
+
+func New(ctx context.Context, opts optypes.CreateOptions) (optypes.Operator, error) {
+	name := opts.Connector.ID.String()
+
+	cred, err := aztypes.GetCredential(opts.Connector.ConfigData)
+	if err != nil {
+		return nil, err
+	}
+
+	return Operator{
+		name:       name,
+		cred:       cred,
+		identifier: hash.SumStrings("azurerm:", cred.SubscriptionID, cred.TenantID, cred.ClientID),
+	}, nil
+}
+
+type Operator struct {
+	name       string
+	cred       *aztypes.Credential
+	identifier string
+}
+
+func (o Operator) Type() optypes.Type {
+	return OperatorType
+}
+
+func (o Operator) IsConnected(ctx context.Context) error {
+	cred, err := azidentity.NewClientSecretCredential(o.cred.TenantID, o.cred.ClientID, o.cred.ClientSecret, nil)
+	if err != nil {
+		return err
+	}
+
+	clientFactory, err := armresources.NewClientFactory(o.cred.SubscriptionID, cred, nil)
+	if err != nil {
+		return err
+	}
+
+	client := clientFactory.NewResourceGroupsClient()
```

Comment:
Is list resource groups the commonly used API for connect check? Does list resource groups require additional permission?

Reply:
Yes. Querying resources of other resource types will need to know the resource group they belong to.